tests/nosetests/pyanaconda_tests/module_source_repo_files_test.py

```diff
@@ -0,0 +1,125 @@
+from pyanaconda.modules.payloads.source.repo_files.initialization import \
+    SetUpRepoFilesSourceTask, TearDownRepoFilesSourceTask
+
+from tests.nosetests.pyanaconda_tests import PropertiesChangedCallback
+
+
+class RepoFilesSourceInterfaceTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.install_media_source_module = RepoFilesSourceModule()
+        self.install_media_source_interface = RepoFilesSourceInterface(
+            self.install_media_source_module
+        )
+
+        self.callback = PropertiesChangedCallback()
+        self.install_media_source_interface.PropertiesChanged.connect(self.callback)
+
+    def type_test(self):
+        """Test Repo files source has a correct type specified."""
+        self.assertEqual(SourceType.REPO_FILES.value, self.install_media_source_interface.Type)
+
+
+class RepoFilesSourceTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.install_media_source_module = RepoFilesSourceModule()
+
+    def type_test(self):
+        """Test Repo files source module has a correct type."""
+        self.assertEqual(SourceType.REPO_FILES, self.install_media_source_module.type)
+
+    def set_up_with_tasks_test(self):
+        """Test Repo files Source set up call."""
+        task_classes = [
+            SetUpRepoFilesSourceTask
+        ]
+
+        # task will not be public so it won't be published
+        tasks = self.install_media_source_module.set_up_with_tasks()
+
+        # Check the number of the tasks
+        task_number = len(task_classes)
+        self.assertEqual(task_number, len(tasks))
+
+        for i in range(task_number):
+            self.assertIsInstance(tasks[i], task_classes[i])
+
+    def tear_down_with_tasks_test(self):
+        """Test Repo files Source ready state for tear down."""
+        task_classes = [
+            TearDownRepoFilesSourceTask
+        ]
+
+        # task will not be public so it won't be published
+        tasks = self.install_media_source_module.tear_down_with_tasks()
+
+        # check the number of tasks
+        task_number = len(task_classes)
+        self.assertEqual(task_number, len(tasks))
+
+        for i in range(task_number):
+            self.assertIsInstance(tasks[i], task_classes[i])
+
+    def ready_state_test(self):
+        """Test Repo files Source ready state for set up."""
+        self.assertTrue(self.install_media_source_module.is_ready())
+
+
+class RepoFilesSourceSetupTaskTestCase(unittest.TestCase):
+
+    def setup_install_source_task_name_test(self):
+        """Test Repo files Source setup installation source task name."""
+        task = SetUpRepoFilesSourceTask("/path/to/mount")
+
+        self.assertEqual(task.name, "Set up Repo files Installation Source")
+
+    @patch("pyanaconda.modules.payloads.source.repo_files.initialization.glob.glob",
+           side_effect=[
+               [], ["/etc/anaconda.repos.d/foo.repo", "/etc/anaconda.repos.d/bar.repo"], [], []
+           ])
```

Comment:
Please create a temp dir with repo files instead. It's much easier for understanding and you will be able to catch problem with glob regex.

Reply:
Done!